evil-common.el

```diff
@@ -1375,48 +1375,43 @@ match-data reflects the last successful match (that caused COUNT
 to reach zero). The behaviour of this functions is similar to
 `up-list'."
   (let* ((count (or count 1))
-         (dir (if (> count 0) +1 -1)))
+         (forwardp (> count 0))
+         (dir (if forwardp +1 -1))
+         (beg+ (if forwardp beg end))
```

Comment:
I wasn't wild about the name either, but I hadn't thought of anything else I thought was particularly better.  I've now replaced the references with the expression itself.